File: sorting_algorithms/selection_sort.rb
# Completed, not refactored

def selection_sort(array_to_sort)
    sorted_array = []
    for i in 0...array_to_sort.length
        @min_found = find_min(array_to_sort)
        sorted_array << @min_found
        array_to_sort.delete_at(array_to_sort.index(array_to_sort.min))    
    end
    sorted_array
end

def find_min(array_to_search)
    @array_min = array_to_search[0]
    if array_to_search.length < 2
        return @array_min
    else
        array_to_search.each_with_index do |elem, inx|
            if elem < @array_min
                @array_min = elem
            end
        end
    end
    @array_min
end

a = (1..25).to_a.shuffle!
b = [3, 6, 5, 7, 9, 8, 7, 3, 5, 4, 7, 9, 9, 8]
selection_sort(a)
selection_sort(b)

File: sorting_algorithms/selection_sort.js
// Completed, not refactored

document.body.innerHTML = "";
function selection_sort(array_to_sort) {
    var sorted_array = [];
    var length_of_array = array_to_sort.length
    for(var i = 0; i < length_of_array; i++){
        var found_min = find_min(array_to_sort);
        sorted_array.push(found_min);
        array_to_sort.splice(array_to_sort.indexOf(found_min), 1);
    }
    return sorted_array;
}

function find_min(array_to_search) {
    var array_min = array_to_search[0];
    var iterations_limit = array_to_search.length-1;
    if(array_to_search.length<2) {
        return array_min;
    } else {
        for(var j = 0; j < iterations_limit; j++) {
            if(array_to_search[(j+1)] < array_min) {
                array_min = array_to_search[(j+1)];
            }
        }
    }
    return array_min;
}

a = [6, 4, 5, 8, 9, 7, 0, 9, 5, 6, 2, 2, 4, 3, 6, 5, 8, 9];
selection_sort(a);